src/rdiff_backup/connection.py

```diff
@@ -588,7 +594,11 @@ def _extract_exception(self):
                 ),
                 log.INFO,
             )
-        return sys.exc_info()[1]
+        result = sys.exc_info()[1]
```

Comment:
This part should be _before_ the call to `robust.is_routine_fatal` because the condition might also raise an OSError, which should be extended with errno_str as well. I don't think it'll make a big difference, because the error is fatal, but just to be consistent.